File: Educational Codeforces Round 86 (Rated for Div. 2)/A. Road To Zero.py
def main():
    x, y = map(int, input().split())
    a, b = map(int, input().split())
    
    if x == y == 0:
        print(0)
        return
    
    ans = min(min(x, y) * b + abs(x-y) * a, (x + y) * a)
    print(ans)
    
if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        main()
        
# https://codeforces.com/problemset/problem/1342/A

File: Educational Codeforces Round 86 (Rated for Div. 2)/A. Road To Zero.cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;

void solve() {
    ll x, y, a, b;
    cin >> x >> y >> a >> b;

    if (x == 0 && y == 0) {
        cout << 0 << "\n";
        return;
    }

    ll ans = min(min(x, y) * b + abs(x - y) * a, (x + y) * a);
    cout << ans << "\n";
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t;
    cin >> t;
    while (t--) {
        solve();
    }

    return 0;
}

// https://codeforces.com/problemset/problem/1342/A